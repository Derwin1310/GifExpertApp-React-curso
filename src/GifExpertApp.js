import {useState} from 'react';
import {AddCategory} from './components/AddCategory';
import {GifGrid} from './components/GifGrid';

const GifExpertApp = ({defaultCategories = []}) => {
	// const [categories, setCategories] = useState(['Attack on Titan']);
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<h2 className='title'>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />

			<ol className='grid-wrapper'>
				{categories.map(category => (
					<GifGrid key={category} category={category} />
				))}
			</ol>
		</>
	);
};

export default GifExpertApp;
