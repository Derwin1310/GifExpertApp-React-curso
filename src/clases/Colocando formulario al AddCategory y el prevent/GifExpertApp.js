import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

//el snippet rafc ya hace todo el functional component por nosotros.
export const GifExpertApp = () => {

    const [categories, setcategories] = useState(['Attack on Titan', 'Demon Slayer', 'Naruto']);
    // const handleAdd = () => setcategories(['HxH',...categories]);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory />
            <hr />
            

            <ol>
                {
                    categories.map(category => {
                        
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </>
    );
};


