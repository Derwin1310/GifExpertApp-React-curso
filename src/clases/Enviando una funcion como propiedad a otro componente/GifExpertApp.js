import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

//el snippet rafc ya hace todo el functional component por nosotros.
export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['Attack on Titan', 'Demon Slayer', 'Naruto']);
    // const handleAdd = () => setCategories(['HxH',...categories]);

    return (
        <>
            <h2>GifExpertApp</h2>
            {/* Aqui le estamos enviando a AddCategory un parametro que es setCategories que es un hook de GEA */}
            <AddCategory setCategories={setCategories} />
            <hr />
            

            <ol>
                {
                    categories.map(category => {
                        
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </>
    );
};


