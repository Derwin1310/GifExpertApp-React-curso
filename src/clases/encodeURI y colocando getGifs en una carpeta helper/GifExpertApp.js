import {useState} from 'react';
import {AddCategory} from './components/AddCategory';
import {GifGrid} from './components/GifGrid';

//el snippet rafc ya hace todo el functional component por nosotros.
export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['Attack on Titan']);

	return (
		<>
			<h2>GifExpertApp</h2>
			<AddCategory setCategories={setCategories} />
			<hr />

			<ol>
				{
					//Aqui preparamos el ol de categories para que devuelva lo que tiene GifGrid que es un h3 de los elementos que vamos a estar insertando en el input del AddCategories y los muestra ahi
					categories.map(category => (
						<GifGrid key={category} category={category} />
					))
				}
			</ol>
		</>
	);
};
