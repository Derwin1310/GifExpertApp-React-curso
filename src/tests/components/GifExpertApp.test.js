import {shallow} from 'enzyme';
import GifExpertApp from '../../GifExpertApp';

describe('Tests in <GifExpertApp />', () => {
	test('should show <GifExpertApp /> correctly with an snapshot', () => {
		const wrapper = shallow(<GifExpertApp />);

		expect(wrapper).toMatchSnapshot();
	});

	test('should show a categories list', () => {
		const categories = ['One Punch', 'Attack on Titan'];
		const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
        console.log(wrapper);

        expect(wrapper).toMatchSnapshot();
        //Aqui hacemos un expect del GifGrid que es el que renderiza las busquedas en el GifExpertApp.
		expect(wrapper.find('GifGrid').length).toBe(categories.length);
	});
});
