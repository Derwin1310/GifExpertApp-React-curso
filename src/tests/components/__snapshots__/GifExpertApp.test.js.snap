// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> should show <GifExpertApp /> correctly with an snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests in <GifExpertApp /> should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Attack on Titan"
      key="Attack on Titan"
    />
  </ol>
</Fragment>
`;
